ricq-core/src/utils/utf8_to_string.rs

```diff
@@ -0,0 +1,16 @@
+/// Optimized `from_utf8_lossy`, presumes that most of buffer is valid utf-8.
+///
+/// The `String::from_utf8_lossy` always allocate memory.
+pub fn utf8_to_string(buf: impl Into<Vec<u8>>) -> String {
+    let buf: Vec<u8> = buf.into(); // for `bytes::Bytes`
```

Comment:
The `Self::Into<Self>` is a no-op.

https://github.com/rust-lang/rust/blob/1.65.0/library/core/src/convert/mod.rs#L556-L565